src/core/system_parameters/beidou_cnav1_navigation_message.cc

```diff
@@ -0,0 +1,605 @@
+	subframe3_string=frame_string.substr(614, BEIDOU_CNAV1_SUBFRAME_3_BITS);
+	
+	std::bitset<BEIDOU_CNAV1_SUBFRAME_1_BITS> subframe1(subframe1_string);  //Gets the message data which contains Subframe 1
+	std::bitset<BEIDOU_CNAV1_SUBFRAME_2_BITS> subframe2(subframe2_string);  //Gets the message data which contains Subframe 2
+	std::bitset<BEIDOU_CNAV1_SUBFRAME_3_BITS> subframe3(subframe3_string);  //Gets the message data which contains Subframe 3
+    
+    
+    	
+    	/* TO DO: Checking CRC for both subframes
+    	//! CNAV1 navigation message have CRC in two subframes i.e in Subframe 2 and Subframe 3
+    	
+    	// Gets the crc data for comparison, last 24 bits from 600 bit of Subframe 2
+    	std::bitset<BEIDOU_CNAV1_CRC_BITS> checksum_sf_2(subframe2.substr(576, 24));
+    	// Gets the crc data for comparison, last 24 bits from 264 bit of Subframe 3
+    	std::bitset<BEIDOU_CNAV1_CRC_BITS> checksum_sf_3(subframe3.substr(240, 24));
+
+    	// Perform data verification and exit code if error in bit sequence
+    	flag_crc_test_sf_2 = crc_test(subframe2, checksum_sf_2.to_ulong());
+    	flag_crc_test_sf_3 = crc_test(subframe3, checksum_sf_3.to_ulong());
+
+    	if (flag_crc_test_sf_2 == false && flag_crc_test_sf_3 == false)
+        	return 0;
+        */
+        
+        // Gets the crc data for comparison, last 24 bits from 288 bit data frame
+    	std::bitset<BEIDOU_CNAV1_CRC_BITS> checksum(frame_string.substr(854, 24));
+
+    	// Perform data verification and exit code if error in bit sequence
+    	flag_crc_test = crc_test(frame_bits, checksum.to_ulong());
+
+    	if (flag_crc_test == false)
+        	return 0;
+    	
+    	//===============================================Decode string message of Subframe 1==============================================
+    	
+    	cnav1_ephemeris.i_satellite_PRN = static_cast<uint32_t>(read_navigation_unsigned(subframe1, PRN));
+    	cnav1_ephemeris.SOH = static_cast<uint32_t>(read_navigation_unsigned(subframe1, SOH)) * 18;   ///Scalling factor= 18, Unit= [s]
+    	flag_TOW_set = true;
+    	flag_TOW_SF_1 = true;
+    	flag_ephemeris_mes_SF1 = true;
+    	
+     	
+    	
+    	//===============================================Decode string message of Subframe 2==============================================
+    	
+    	cnav1_ephemeris.i_BDS_week = static_cast<double>(read_navigation_unsigned(subframe2, WN_S_2));  //[week] effective range 0~8191
+    	cnav1_ephemeris.HOW = static_cast<double>(read_navigation_unsigned(subframe2, HOW_S_2));       //[s] effective range 0~604797
+    	cnav1_ephemeris.IODC = static_cast<double>(read_navigation_unsigned(subframe2, IODC_S_2));
+    	cnav1_ephemeris.IODE = static_cast<double>(read_navigation_unsigned(subframe2, IODE_S_2));
+    	
+        // Ephemeris I Start
+        cnav1_ephemeris.t_oe = static_cast<double>(read_navigation_unsigned(subframe2, t_oe_S_2)) * 300;                              //[s] effective range 0~604500
```

Comment:
Make this a constant rather than a hardcoded number

Reply:
Replaced hardcoded number by constant.